lib/src/ruby/module.cc

```
@@ -116,11 +116,12 @@ extern "C" {
         set_level(log_level::warning);
 
         // Initialize ruby
-        auto ruby = api::instance();
-        if (!ruby) {
-            return;
+        try {
+            auto& ruby = api::instance();
+            ruby.initialize();
+        } catch (runtime_error ex) {
+            LOG_WARNING("%1%: facts requiring Ruby will not be resolved.", ex.what());
```

If an exception is thrown, we should return rather than continuing.